src/libraries/Common/src/System/HexConverter.cs

```diff
@@ -288,8 +288,42 @@ public static int FromLowerChar(int c)
         }
 
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        public static bool IsHexChar(int c)
+        public static unsafe bool IsHexChar(int c)
         {
+            if (sizeof(IntPtr) == 8)
```

Comment:
since corelib is arch specific I guess it's better (e.g. for JIT) to use `#if TARGET_64BIT` here

Reply:
This is in the Common folder and is pulled in by a bunch of different libs, some of which only build AnyCPU. If we wanted an ifdef for the target platform, we'd also need to ifdef CORELIB. (See _ValueStringBuilder.cs_ for some examples.)

Reply:
@GrabYourPitchforks ah, good point!